### DISCUSSION TITLE
const CustomLink

### DISCUSSION THREAD

**Discussion by rsipakov** [timlrx/tailwind-nextjs-starter-blog#329] [Q&A] [answered]
Hi All! 
Is it possible to rename file '/components/Link.js' to 'components/CustomLink.js' without affecting the code as a whole? I think it's easier to orient and not confuse it with the general `import Link from 'next/link`. 
Thank you!

**Comment by timlrx** [comment, marked as answer]
Yes, no issues at all. You just need to rename all the imports that reference the Link component as well (VScode automatically does it). 

If you are thinking about making the change on the main template, I still prefer the `Link.js` rather than `CustomLink.js`. The main idea is that you should have one link component and there's no need to use `next/link` directly. I find it easier to customise the link / a tag logic this way.

**Reply by rsipakov** [reply]
In this case why used 

```
import Link from '@/components/Link'

<Link
   href={`/blog/${slug}`}
   className="text-gray-900 dark:text-gray-100">
   {title}
</Link>
``` 

Instead of 

```
import CustomLink from '@/components/Link'

<CustomLink
   href={`/blog/${slug}`}
   className="text-gray-900 dark:text-gray-100">
   {title}
</CustomLink>
```
if we have

`export default CustomLink` ?

Thank you!

**Reply by timlrx** [reply]
`export default CustomLink` is just to avoid a name clash with `next/link` which I believe is imported as `Link`. Subsequently, it's just easier to refer to it as `Link`. I could have imported `next/Link` as `NextLink` and exported the main one as `Link`. It's just a matter of names :)

**Reply by rsipakov** [reply]
Thank you @timlrx! One more question, why for the Tag component used regular `import Link from next/link` if for other components used custom `import Link from '@/components/Link'`?

**Reply by timlrx** [reply]
Good catch! Will change this.

**Comment by rsipakov** [comment]
I agree with your explanation, this one is more logical. Thank you!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
